self_play.py

```diff
@@ -220,8 +230,11 @@ def play_game(model1, model2, mcts_simulations, stop_exploration, self_play=Fals
             if self_play:
                 other_mcts = mcts_tree
 
+        if disable_resign:
+            skip_resign=not game_final(get_real_board(board))
```

Comment:
I'm not sure I understand game_final's purpose. 

Is is to prevent passing at random times and "force" the algorithm to play until the board is full of stones ?